openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,516 @@
+        description='featureFlags defines the list of features to enable for Velero instance',
+    )
+    logLevel: Optional[LogLevel] = Field(
+        None,
+        description='Velero server’s log level (use debug for the most logging, leave unset for velero default)',
+    )
+    noDefaultBackupLocation: Optional[bool] = Field(
+        None,
+        description='If you need to install Velero without a default backup storage location noDefaultBackupLocation flag is required for confirmation',
+    )
+    podConfig: Optional[DataProtectionApplicationConfigurationVeleroPodConfig] = Field(
+        None, description='Pod specific configuration'
+    )
+    restoreResourcesVersionPriority: Optional[str] = Field(
+        None,
+        description='restoreResourceVersionPriority represents a configmap that will be created if defined for use in conjunction with EnableAPIGroupVersions feature flag Defining this field automatically add EnableAPIGroupVersions to the velero server feature flag',
+    )
+
+
+class DataProtectionApplicationConfiguration(BaseModel):
+    restic: Optional[Restic] = Field(
+        None, description='ResticConfig is the configuration for restic server'
+    )
+    velero: Optional[DataProtectionApplicationConfigurationVelero] = None
+
+
+class DataMover(BaseModel):
+    credentialName: Optional[str] = Field(
+        None, description='User supplied Restic Secret name'
+    )
+    enable: Optional[bool] = Field(
+        None,
+        description='enable flag is used to specify whether you want to deploy the volume snapshot mover controller',
+    )
+    timeout: Optional[str] = Field(
+        None,
+        description='User supplied timeout to be used for VolumeSnapshotBackup and VolumeSnapshotRestore to complete, default value is 10m',
+    )
+
+
+class Features(BaseModel):
+    dataMover: Optional[DataMover] = Field(
+        None, description='Contains data mover specific configurations'
+    )
+
+
+class Option(BaseModel):
+    name: Optional[str] = Field(None, description='Required.')
+    value: Optional[str] = None
+
+
+class PodDnsConfig(BaseModel):
+    nameservers: Optional[List[str]] = Field(
+        None,
+        description='A list of DNS name server IP addresses. This will be appended to the base nameservers generated from DNSPolicy. Duplicated nameservers will be removed.',
+    )
+    options: Optional[List[Option]] = Field(
+        None,
+        description='A list of DNS resolver options. This will be merged with the base options generated from DNSPolicy. Duplicated entries will be removed. Resolution options given in Options will override those that appear in the base DNSPolicy.',
+    )
+    searches: Optional[List[str]] = Field(
+        None,
+        description='A list of DNS search domains for host-name lookup. This will be appended to the base search paths generated from DNSPolicy. Duplicated search paths will be removed.',
+    )
+
+
+class SnapshotLocationVelero(BaseModel):
```

Comment:
please rename SnapshotLocationVelero -> VolumeSnapshotLocationSpec

Also, maybe we should move this to a new resource volume_snapshot_location.py

Reply:
waiting for PR#31

Reply:
it's done: https://github.com/oadp-qe/openshift-adp-python-wrapper/pull/31 please review

Reply:
approved #31

Reply:
@mperetzred